java/src/main/java/com/saucelabs/saucebindings/SauceOptions.java

```diff
@@ -289,6 +427,23 @@ private void setEnumCapability(String key, String value) {
                     setBrowserName(Browser.valueOf(Browser.fromString(value)));
                 }
                 break;
+            case "deviceOrientation":
+                if (!DeviceOrientation.keys().contains(value)) {
```

Comment:
Wow, this method is getting scary! I created a PR to clean it up https://github.com/saucelabs/sauce_bindings/pull/171